traits/trait_handlers.py

```diff
@@ -2956,7 +2928,7 @@ def __deepcopy__(self, memo):
             self.trait,
             lambda: None,
             self.name,
-            dict([copy.deepcopy(x, memo) for x in six.iteritems(self)]),
+            dict([copy.deepcopy(x, memo) for x in self.items()]),
```

Comment:
Drive-by fix: is the intermediate list actually needed here?

Reply:
Very likely not.
```suggestion
            dict(copy.deepcopy(x, memo) for x in self.items()),
```